### PR TITLE
[PM-163] add db for mock-psp

### PR DESCRIPTION
**Pull request by infantesimone**
### List of changes

- add `postgresql` db for `mock-psp`

### Motivation and context

The goal is to create a `postgresql` database for the `mock-psp`
 
### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

**Comment by pasqualedevita** [comment]
this postgres will be used only by mock?

File: src/database.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  subnet_id                        = module.db_snet.id
+  administrator_login              = data.azurerm_key_vault_secret.db_administrator_login.value
+  administrator_login_password     = data.azurerm_key_vault_secret.db_administrator_login_password.value
+  sku_name                         = var.db_sku_name
+  storage_mb                       = var.db_storage_mb
+  db_version                       = 10
+  geo_redundant_backup_enabled     = var.db_geo_redundant_backup_enabled
+  enable_replica                   = var.db_enable_replica
+  ssl_minimal_tls_version_enforced = "TLS1_2"
+  public_network_access_enabled    = true
```
Comment (pasqualedevita) [inline src/database.tf:34]:
```suggestion
  public_network_access_enabled    = false
```

File: src/database.tf [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  name         = "db-administrator-login"
+  key_vault_id = module.key_vault.id
+}
+
+data "azurerm_key_vault_secret" "db_administrator_login_password" {
+  name         = "db-administrator-login-password"
+  key_vault_id = module.key_vault.id
+}
+
+#tfsec:ignore:azure-database-no-public-access
```
Comment (pasqualedevita) [inline src/database.tf:18]:
```suggestion
```
no need

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Comment by pasqualedevita** [comment]
```
If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
var.db_sku_name
  Specifies the SKU Name for this PostgreSQL Server
```
check vars, plan it's blocked because seems is missing `var.db_sku_name`

**Comment by infantesimone** [comment]
> this postgres will be used only by mock?

yes

File: src/env/dev/terraform.tfvars [hunk 3]
```diff
@@ -39,3 +40,7 @@ app_gateway_management_certificate_name = "management-dev-platform-pagopa-it"
 mock_ec_enabled = true
 
 mock_psp_enabled = true
+
+db_sku_name = "B_Gen4_1"
```
Comment (infantesimone) [inline src/env/dev/terraform.tfvars:44]: @pasqualedevita according to [this](https://azure.microsoft.com/en-us/pricing/details/postgresql/server/), B_Gen4_1 is the cheapest postgres sku? suitable for mocks?

Comment (pasqualedevita) [reply src/env/dev/terraform.tfvars:44]: should be `B_Gen5_1`

ref https://docs.microsoft.com/it-it/cli/azure/postgres/server?view=azure-cli-latest#az_postgres_server_list_skus 

`az postgres server list-skus -l westeurope`

```
[
  {
    "features": [],
    "id": "Basic",
    "maxBackupRetentionDays": 35,
    "maxLargeStorageMb": 0,
    "maxStorageMb": 1048576,
    "minBackupRetentionDays": 7,
    "minLargeStorageMb": 0,
    "minStorageMb": 5120,
    "serviceLevelObjectives": [
      {
        "edition": "Basic",
        "hardwareGeneration": "Gen5",
        "id": "B_Gen5_1",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 1048576,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 1
      },
      {
        "edition": "Basic",
        "hardwareGeneration": "Gen5",
        "id": "B_Gen5_2",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 1048576,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 2
      }
    ],
    "subscriptionFeatures": []
  },
  {
    "features": [
      {
        "name": "MicrosoftDBPostgreSQLBYOK",
        "value": "Enabled"
      },
      {
        "name": "MicrosoftDBPostgreSQLInfrastructureEncryption",
        "value": "Enabled"
      }
    ],
    "id": "GeneralPurpose",
    "maxBackupRetentionDays": 35,
    "maxLargeStorageMb": 0,
    "maxStorageMb": 16777216,
    "minBackupRetentionDays": 7,
    "minLargeStorageMb": 0,
    "minStorageMb": 5120,
    "serviceLevelObjectives": [
      {
        "edition": "GeneralPurpose",
        "hardwareGeneration": "Gen5",
        "id": "GP_Gen5_2",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 16777216,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 2
      },
      {
        "edition": "GeneralPurpose",
        "hardwareGeneration": "Gen5",
        "id": "GP_Gen5_4",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 16777216,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 4
      },
      {
        "edition": "GeneralPurpose",
        "hardwareGeneration": "Gen5",
        "id": "GP_Gen5_8",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 16777216,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 8
      },
      {
        "edition": "GeneralPurpose",
        "hardwareGeneration": "Gen5",
        "id": "GP_Gen5_16",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 16777216,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 16
      },
      {
        "edition": "GeneralPurpose",
        "hardwareGeneration": "Gen5",
        "id": "GP_Gen5_32",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 16777216,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 32
      },
      {
        "edition": "GeneralPurpose",
        "hardwareGeneration": "Gen5",
        "id": "GP_Gen5_64",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 16777216,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 64
      }
    ],
    "subscriptionFeatures": []
  },
  {
    "features": [
      {
        "name": "MicrosoftDBPostgreSQLBYOK",
        "value": "Enabled"
      },
      {
        "name": "MicrosoftDBPostgreSQLInfrastructureEncryption",
        "value": "Enabled"
      }
    ],
    "id": "MemoryOptimized",
    "maxBackupRetentionDays": 35,
    "maxLargeStorageMb": 0,
    "maxStorageMb": 16777216,
    "minBackupRetentionDays": 7,
    "minLargeStorageMb": 0,
    "minStorageMb": 5120,
    "serviceLevelObjectives": [
      {
        "edition": "MemoryOptimized",
        "hardwareGeneration": "Gen5",
        "id": "MO_Gen5_2",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 16777216,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 2
      },
      {
        "edition": "MemoryOptimized",
        "hardwareGeneration": "Gen5",
        "id": "MO_Gen5_4",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 16777216,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 4
      },
      {
        "edition": "MemoryOptimized",
        "hardwareGeneration": "Gen5",
        "id": "MO_Gen5_8",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 16777216,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 8
      },
      {
        "edition": "MemoryOptimized",
        "hardwareGeneration": "Gen5",
        "id": "MO_Gen5_16",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 16777216,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 16
      },
      {
        "edition": "MemoryOptimized",
        "hardwareGeneration": "Gen5",
        "id": "MO_Gen5_32",
        "maxBackupRetentionDays": 35,
        "maxStorageMb": 16777216,
        "minBackupRetentionDays": 7,
        "minStorageMb": 5120,
        "vCore": 32
      }
    ],
    "subscriptionFeatures": []
  }
]
```

**Comment by pasqualedevita** [comment]
@infantesimone u have to to an `update branch`

**Review by pasqualespica — DISMISSED** [review]
lgtm 👍

File: src/env/dev/terraform.tfvars [hunk 4]
```diff
@@ -14,6 +14,7 @@ cidr_subnet_appgateway = ["10.1.128.0/24"]
 cidr_subnet_azdoa      = ["10.1.130.0/24"]
 cidr_subnet_mock_ec    = ["10.1.240.0/29"]
 cidr_subnet_mock_psp   = ["10.1.240.8/29"]
+cidr_subnet_postgresql = ["10.1.240.16/29"]
```
Comment (pasqualedevita) [inline src/env/dev/terraform.tfvars:17]:
```suggestion
cidr_subnet_postgresql = ["10.1.129.0/24"]
```

File: src/env/uat/terraform.tfvars [hunk 5]
```diff
@@ -14,6 +14,7 @@ cidr_subnet_appgateway = ["10.1.128.0/24"]
 cidr_subnet_azdoa      = ["10.1.130.0/24"]
 cidr_subnet_mock_ec    = ["10.1.240.0/29"]
 cidr_subnet_mock_psp   = ["10.1.240.8/29"]
+cidr_subnet_postgresql = ["10.1.240.16/29"]
```
Comment (pasqualedevita) [inline src/env/uat/terraform.tfvars:17]:
```suggestion
cidr_subnet_postgresql = ["10.1.129.0/24"]
```

File: src/env/uat/terraform.tfvars [hunk 6]
```diff
@@ -37,4 +38,8 @@ app_gateway_management_certificate_name = "management-uat-platform-pagopa-it"
 app_gateway_min_capacity                = 1
 app_gateway_max_capacity                = 2
 
-mock_ec_enabled = true
+mock_psp_enabled                 = false
+db_sku_name                      = "B_Gen5_1"
+db_enable_replica                = false
+db_public_network_access_enabled = false
+prostgres_enabled                = false
```
Comment (pasqualedevita) [inline src/env/uat/terraform.tfvars:45]:
```suggestion
postgresql_sku_name                      = "GP_Gen5_2" # fixme verify
postgresql_enable_replica                = false
postgresql_public_network_access_enabled = true
prostgresql_enabled                      = false
postgresql_network_rules = {
  ip_rules = [
    "0.0.0.0/0"
  ]
  # dblink
  allow_access_to_azure_services = false
}
prostgresql_db_mockpsp = "mock-psp" 
```

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**